src/serverenvironment.cpp

```diff
@@ -484,27 +484,25 @@ bool ServerEnvironment::removePlayerFromDatabase(const std::string &name)
 bool ServerEnvironment::line_of_sight(v3f pos1, v3f pos2, float stepsize, v3s16 *p)
 {
 	float distance = pos1.getDistanceFrom(pos2);
-
-	//calculate normalized direction vector
-	v3f normalized_vector = v3f((pos2.X - pos1.X)/distance,
-		(pos2.Y - pos1.Y)/distance,
-		(pos2.Z - pos1.Z)/distance);
-
-	//find out if there's a node on path between pos1 and pos2
-	for (float i = 1; i < distance; i += stepsize) {
-		v3s16 pos = floatToInt(v3f(normalized_vector.X * i,
-			normalized_vector.Y * i,
-			normalized_vector.Z * i) +pos1,BS);
-
-		MapNode n = getMap().getNodeNoEx(pos);
-
-		if(n.param0 != CONTENT_AIR) {
-			if (p) {
-				*p = pos;
-			}
+	// For compatibility, reduce the length of the line
+	// by stepsize, keeping the end's position
+	if (distance < stepsize) {
+		return true;
```

Comment:
Some mods might use this function in a loop, as:

```lua
pos1 = {x = 0, y = 0, z = 0}
pos1 = {x = 10, y = 10, z = 10}
repeat
    found, pos1 = minetest.line_of_sight(pos1, pos2, 0.1)
until found and <pos1 is not glass>
```

This won't run for an infinite time because `line_of_sight` always steps at least one step size. I kept that behavior to avoid creating infinite loops.